strapi/strapi
# api-token-v2/rbac-on-api-token-list

What does it do?
As an admin user who can manage RBAC, I want to define permissions so that access to the API tokens list can be managed and limited.

How to test it?
Should satisfy the following criterias:

- As an admin user accessing the API tokens list
   -  when I don’t have any permission
      - then I have the “Create token” button not accessible and the lists of tokens  hidden with the message “You don’t have the permission to access that content”
[https://www.figma.com/file/rtPKsszPVpuoS91ZUMnlcp/Design-System---Business-compo-only?node-id=145%3A30282 - Connetti ad anteprima](https://www.figma.com/file/rtPKsszPVpuoS91ZUMnlcp/Design-System---Business-compo-only?node-id=145%3A30282)
   -  when I have **only** the Create permission
      - then I have the “Create token” button  accessible and the lists of tokens  hidden with the message “You don’t have the permission to access that content”
[https://www.figma.com/file/rtPKsszPVpuoS91ZUMnlcp/Design-System---Business-compo-only?node-id=145%3A30282 - Connetti ad anteprima](https://www.figma.com/file/rtPKsszPVpuoS91ZUMnlcp/Design-System---Business-compo-only?node-id=145%3A30282) 
   - when I have **only** the Read permission
     - then I have the lists of tokens to be accessible but with theUpdate and Delete commands for every token in the list not accessible (hidden) and the Create button hidden as well
   - when I have **only** the Update permission
     - then it doesn’t have effects on the list of tokens (as long as the Read permission is not active, I can’t get the list of tokens to select one and edit it)
   - when I have **only** the Delete permission
     - then it doesn’t have effects on the list of tokens (as long as the Read permission is not active, I can’t get the list of tokens to select one and edit it)
   - when I have a combination of the permissions applied
     - then the consequences deriving from these permissions are applied, in particular
       - when I have the Read and Update permission (but not the Delete permission)
         - then I have the list of tokens with the edit command accessible (and with the delete command not accessible)
       - when I have the Read and Delete permissions (but not the Update permission)
         - then I have the list of tokens with the update and delete command accessible [ we will consider to replace the pencil icon - that makes the user think that he can edit the token - with an eye icon, since in this case the user can just read the detail in the list of tokens as well as in the single API token settings]
       - when I have the Delete and Update permissions (but not the Read permission)
         - then I can’t see the list of tokens and so I neither have access to the update and delete commands (as long as you don’t also have the read permission, anything can be displayed and so edited), so I have the “Create token” button hidden and the lists of tokens  hidden with the message “You don’t have the permission to access that content”
[https://www.figma.com/file/rtPKsszPVpuoS91ZUMnlcp/Design-System---Business-compo-only?node-id=145%3A30282 - Connetti ad anteprima](https://www.figma.com/file/rtPKsszPVpuoS91ZUMnlcp/Design-System---Business-compo-only?node-id=145%3A30282) 
       - when I have the Read, Update, and Delete permissions
         - then I have the list of tokens with the edit and the delete command accessible
